scipy/stats/_continuous_distns.py

```diff
@@ -3616,6 +3616,50 @@ def _cdf(self, x, mu):
     def _stats(self, mu):
         return mu, mu**3.0, 3*np.sqrt(mu), 15*mu
 
+    def fit(self, data, *args, **kwds):
+
+        if type(self) == wald_gen:
+            return super(invgauss_gen, self).fit(data, *args, **kwds)
+
+        floc = kwds.pop('floc', None)
+        fscale = kwds.pop('fscale', None)
+        # borrowed from distn_infrastructure
+        # puts shape into kwds['f0']
+        if self.shapes:
+            shapes = self.shapes.replace(',', ' ').split()
+            for j, s in enumerate(shapes):
+                key = 'f' + str(j)
+                names = [key, 'f' + s, 'fix_' + s]
+                val = _get_fixed_fit_value(kwds, names)
+                if val is not None:
+                    kwds[key] = val
+        f0_s = kwds.pop('f0', None)
+        data = np.asarray(data)
+        _check_fit_input_parameters(data, args, kwds, fixed_param=(floc, fscale, f0_s))
+        '''
+        MLE is not used in 3 condtions:
```

Comment:
```suggestion
        Analytical MLE is not used in 3 conditions:
```
Both the analytical formulas and the optimizer are doing MLE "Maximum Likelihood Estimation".